### ISSUE TITLE
[syzkaller] WARNING in mptcp_reset_timer

### ISSUE THREAD

**Issue by cpaasch** [multipath-tcp/mptcp_net-next#126]
```
------------[ cut here ]------------
WARNING: CPU: 1 PID: 12512 at net/mptcp/protocol.c:761 mptcp_reset_timer+0x12a/0x160 net/mptcp/protocol.c:761
Modules linked in:
CPU: 1 PID: 12512 Comm: kworker/1:6 Not tainted 5.10.0-rc6 #52
Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS rel-1.13.0-0-gf21b5a4aeb02-prebuilt.qemu.org 04/01/2014
Workqueue: events mptcp_worker
RIP: 0010:mptcp_reset_timer+0x12a/0x160 net/mptcp/protocol.c:761
Code: e8 4b 0c ad ff e8 56 21 88 fe 48 b8 00 00 00 00 00 fc ff df 48 c7 04 03 00 00 00 00 48 83 c4 40 5b 5d 41 5c c3 e8 36 21 88 fe <0f> 0b 41 bc c8 00 00 00 eb 98 e8 e7 b1 af fe e9 30 ff ff ff 48 c7
RSP: 0018:ffffc900018c7c68 EFLAGS: 00010293
RAX: ffff888108cb1c80 RBX: 1ffff92000318f8d RCX: ffffffff82ad0307
RDX: 0000000000000000 RSI: ffffffff82ad036a RDI: 0000000000000007
RBP: ffff888113e2d000 R08: ffff888108cb1c80 R09: ffffed10227c5ab7
R10: ffff888113e2d5b7 R11: ffffed10227c5ab6 R12: 0000000000000000
R13: ffff88801f100000 R14: ffff888113e2d5b0 R15: 0000000000000001
FS:  0000000000000000(0000) GS:ffff88811b500000(0000) knlGS:0000000000000000
CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
CR2: 00007fd76a874ef8 CR3: 000000001689c005 CR4: 0000000000170ee0
Call Trace:
 mptcp_worker+0xaa4/0x1560 net/mptcp/protocol.c:2334
 process_one_work+0x8d3/0x1200 kernel/workqueue.c:2272
 worker_thread+0x9c/0x1090 kernel/workqueue.c:2418
 kthread+0x303/0x410 kernel/kthread.c:292
 ret_from_fork+0x22/0x30 arch/x86/entry/entry_64.S:296
```

HEAD:
05cb27b258ba ("DO-NOT-MERGE: mptcp: enabled by default")  (HEAD, tag: export/20201209T060936, mptcp_net-next/export) (12 hours ago) <Matthieu Baerts>
525593cf6729 ("DO-NOT-MERGE: mptcp: add GitHub Actions")  (12 hours ago) <Matthieu Baerts>
6aa8731bf964 ("DO-NOT-MERGE: mptcp: use kmalloc on kasan build")  (12 hours ago) <Paolo Abeni>
2227bfd2f6bd ("mptcp: let MPTCP create max size skbs")  (12 hours ago) <Paolo Abeni>
908c6322325f ("mptcp: pm: simplify select_local_address()")  (12 hours ago) <Paolo Abeni>
a771b76cf445 ("mptcp: parse and act on incoming FASTCLOSE option")  (12 hours ago) <Florian Westphal>
7dbc6b7828a9 ("tcp: parse mptcp options contained in reset packets")  (12 hours ago) <Florian Westphal>
4598a67e2407 ("mptcp: hold mptcp socket before calling tcp_done")  (12 hours ago) <Florian Westphal>
363050070e94 ("mptcp: use MPTCPOPT_HMAC_LEN macro")  (12 hours ago) <Geliang Tang>
905c00c9fd2b ("selftests: mptcp: add the flush addrs testcase")  (12 hours ago) <Geliang Tang>
2d0de9b2d12d ("mptcp: remove address when netlink flushes addrs")  (12 hours ago) <Geliang Tang>
389cb8de6149 ("mptcp: use the variable sk instead of open-coding")  (12 hours ago) <Geliang Tang>
62ad6da7d479 ("mptcp: rename add_addr_signal and mptcp_add_addr_status")  (12 hours ago) <Geliang Tang>
56607a997374 ("mptcp: drop rm_addr_signal flag")  (12 hours ago) <Geliang Tang>
f5614981a113 ("mptcp: print out port and ahmac when receiving ADD_ADDR")  (12 hours ago) <Geliang Tang>
faec918a7bb5 ("mptcp: add port parameter for mptcp_pm_announce_addr")  (12 hours ago) <Geliang Tang>
1bab32fc5774 ("mptcp: send out dedicated packet for ADD_ADDR using port")  (12 hours ago) <Geliang Tang>
a7429bbd1f46 ("mptcp: add the outgoing ADD_ADDR port support")  (12 hours ago) <Geliang Tang>
a8787a82b5a1 ("mptcp: use adding up size to get ADD_ADDR length")  (12 hours ago) <Geliang Tang>
1690597e3a47 ("mptcp: add port support for ADD_ADDR suboption writing")  (12 hours ago) <Geliang Tang>
4021cd88160f ("mptcp: unify ADD_ADDR and ADD_ADDR6 suboptions writing")  (12 hours ago) <Geliang Tang>
0b863095033d ("mptcp: unify ADD_ADDR and echo suboptions writing")  (12 hours ago) <Geliang Tang>
c855f8921120 ("bpf:selftests: add bpf_mptcp_sock() verifier tests")  (12 hours ago) <Nicolas Rybowski>
0eaea54e6080 ("bpf:selftests: add MPTCP test base")  (12 hours ago) <Nicolas Rybowski>
eed59abf1b73 ("bpf: add 'bpf_mptcp_sock' structure and helper")  (12 hours ago) <Nicolas Rybowski>
6dd1da955adf ("mptcp: attach subflow socket to parent cgroup")  (12 hours ago) <Nicolas Rybowski>
58a4d0cfd053 ("bpf: expose is_mptcp flag to bpf_tcp_sock")  (12 hours ago) <Nicolas Rybowski>
d188dfee6155 ("mptcp: be careful on subflows shutdown")  (12 hours ago) <Paolo Abeni>
99102016f674 ("mptcp: plug subflow context memory leak")  (12 hours ago) <Paolo Abeni>
ae1cd5eb674c ("mptcp: link MPC subflow into msk only after accept")  (12 hours ago) <Paolo Abeni>
afae3cc2da10 ("net: atheros: simplify the return expression of atl2_phy_setup_autoneg_adv()")  (mptcp_net-next/net-next) (18 hours ago) <Zheng Yongjun>

No reproducer.

CONFIG-file:
[CONFIG.txt](https://github.com/multipath-tcp/mptcp_net-next/files/5674167/CONFIG.txt)

**Comment by matttbe** [comment]
might be linked to #105 (still happening) and #70

**Comment by dcaratti** [comment]
ATM the syz repro is causing a splat similar to the one in issue #115 , will fix this first

**Comment by pabeni** [comment]
@cpaasch: on current export branch this one should at least trigger a different back-trace and possibly could be fixed, could you still reproduce it?

**Comment by cpaasch** [comment]
Not happening right now. If I see it again, I will update you here.

**Comment by matttbe** [comment]
Thx for having tested!

I suggest to close this ticket, feel free to re-open it if needed.

**Comment by cpaasch** [comment]
Reopening...

HEAD is at:
b4875a2a7e4a ("some more debugging in sk_stream_kill_queues")  (HEAD) (67 minutes ago) <Christoph Paasch>
70856533a4d9 ("DO-NOT-MERGE: mptcp: enabled by default")  (tag: export/20210128T064336, mptcp_net-next/export) (12 hours ago) <Matthieu Baerts>
a475850233b6 ("DO-NOT-MERGE: mptcp: add GitHub Actions")  (12 hours ago) <Matthieu Baerts>
489c73c74130 ("DO-NOT-MERGE: mptcp: use kmalloc on kasan build")  (12 hours ago) <Paolo Abeni>
3519fa325557 ("mptcp: add netlink event support")  (12 hours ago) <Florian Westphal>
6db4b0635a33 ("genetlink: add CAP_NET_ADMIN test for multicast bind")  (12 hours ago) <Florian Westphal>
6c950820c9c5 ("mptcp: avoid lock_fast usage in accept path")  (12 hours ago) <Florian Westphal>
5c072ef9aed0 ("mptcp: pass subflow socket to a few helpers")  (12 hours ago) <Florian Westphal>
2709acb7cea6 ("mptcp: split __mptcp_close_ssk helper")  (12 hours ago) <Florian Westphal>
ad9540369a8f ("mptcp: move pm netlink work into pm_netlink")  (12 hours ago) <Florian Westphal>
b77a832e09ee ("mptcp: pm: add lockdep assertions")  (12 hours ago) <Florian Westphal>
3d2aa9057623 ("selftests: mptcp: add command line arguments for mptcp_join.sh")  (12 hours ago) <Geliang Tang>
3105f3e7435b ("selftests: mptcp: add testcases for ADD_ADDR with port")  (12 hours ago) <Geliang Tang>
53118be7c75a ("mptcp: add the mibs for ADD_ADDR with port")  (12 hours ago) <Geliang Tang>
fe8cb87a14c8 ("selftests: mptcp: add port argument for pm_nl_ctl")  (12 hours ago) <Geliang Tang>
e5190ee5ecc5 ("mptcp: deal with MPTCP_PM_ADDR_ATTR_PORT in PM netlink")  (12 hours ago) <Geliang Tang>
6322ac0958d8 ("mptcp: enable use_port when invoke addresses_equal")  (12 hours ago) <Geliang Tang>
631325c84980 ("mptcp: add port number check for MP_JOIN")  (12 hours ago) <Geliang Tang>
c06d0861a72d ("mptcp: add a new helper subflow_req_create_thmac")  (12 hours ago) <Geliang Tang>
99be95715a1d ("mptcp: drop unused skb in subflow_token_join_request")  (12 hours ago) <Geliang Tang>
12a982e474d2 ("mptcp: create the listening socket for new port")  (12 hours ago) <Geliang Tang>
9147162e1048 ("selftests: mptcp: add testcases for newly added addresses")  (12 hours ago) <Geliang Tang>
4010c2da6bf6 ("selftests: mptcp: use minus values for removing address numbers")  (12 hours ago) <Geliang Tang>
ff8b5021e022 ("mptcp: send ack for every add_addr")  (12 hours ago) <Geliang Tang>
cc6454e76894 ("mptcp: create subflow or signal addr for newly added address")  (12 hours ago) <Geliang Tang>
b37ae55190ee ("mptcp: drop *_max fields in mptcp_pm_data")  (12 hours ago) <Geliang Tang>
8aa9be3366cc ("mptcp: use WRITE_ONCE/READ_ONCE for the pernet *_max")  (12 hours ago) <Geliang Tang>
cae4ced4d374 ("bpf:selftests: add bpf_mptcp_sock() verifier tests")  (12 hours ago) <Nicolas Rybowski>
7bd6ee3bc08e ("bpf:selftests: add MPTCP test base")  (12 hours ago) <Nicolas Rybowski>
d36880e9f98e ("bpf: add 'bpf_mptcp_sock' structure and helper")  (12 hours ago) <Nicolas Rybowski>
710910e564d3 ("bpf: expose is_mptcp flag to bpf_tcp_sock")  (12 hours ago) <Nicolas Rybowski>
75a7ca5b9d49 ("linux: handle MPTCP consistently with TCP")  (12 hours ago) <Paolo Abeni>
d1f3bdd4eaae ("net: dsa: rtl8366rb: standardize init jam tables")  (mptcp_net-next/net-next) (14 hours ago) <Lorenzo Carletti>

```
RBP: 0000000000000102 R08: 0000000000000000 R09: 0000000000000000
R10: 0000000000000000 R11: 0000000000000246 R12: 000000000069bf6c
R13: 00007ffdd8439abf R14: 00007f4aa5ea3000 R15: 0000000000000003
------------[ cut here ]------------
WARNING: CPU: 1 PID: 10368 at net/mptcp/protocol.c:764 mptcp_reset_timer+0x12a/0x160 net/mptcp/protocol.c:764
Modules linked in:
CPU: 1 PID: 10368 Comm: kworker/1:2 Not tainted 5.11.0-rc4b4875a2a7e4a3a770cb007c5a294bdb7a647adf0 #66
Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS rel-1.13.0-0-gf21b5a4aeb02-prebuilt.qemu.org 04/01/2014
Workqueue: events mptcp_worker
RIP: 0010:mptcp_reset_timer+0x12a/0x160 net/mptcp/protocol.c:764
Code: e8 7b 2b ac ff e8 a6 2e 83 fe 48 b8 00 00 00 00 00 fc ff df 48 c7 04 03 00 00 00 00 48 83 c4 40 5b 5d 41 5c c3 e8 86 2e 83 fe <0f> 0b 41 bc c8 00 00 00 eb 98 e8 27 fc aa fe e9 30 ff ff ff 48 c7
RSP: 0018:ffffc90003b1fc60 EFLAGS: 00010293
RAX: 0000000000000000 RBX: 1ffff92000763f8c RCX: 0000000000000000
RDX: ffff888030f6c9c0 RSI: ffffffff82b2bf5a RDI: 0000000000000003
RBP: ffff8880136f7000 R08: 0000000000000000 R09: 0000000000000001
R10: ffffffff82b2bef7 R11: 0000000000000007 R12: 0000000000000000
R13: ffff8880136f75b8 R14: ffff888015970000 R15: 0000000000000001
FS:  0000000000000000(0000) GS:ffff88811b500000(0000) knlGS:0000000000000000
CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
CR2: 00007ffddac74c4c CR3: 000000010a73c006 CR4: 0000000000170ee0
Call Trace:
 mptcp_worker+0xe47/0x1310 net/mptcp/protocol.c:2307
 process_one_work+0x896/0x1170 kernel/workqueue.c:2275
 worker_thread+0x605/0x1350 kernel/workqueue.c:2421
 kthread+0x344/0x410 kernel/kthread.c:292
 ret_from_fork+0x22/0x30 arch/x86/entry/entry_64.S:296
```

CONFIG-file:
[CONFIG.txt](https://github.com/multipath-tcp/mptcp_net-next/files/5889385/CONFIG.txt)

**Comment by cpaasch** [comment]
Actually - no ! This is the same as sk_stream_kill_queues...

**Comment by cpaasch** [comment]
Reopening - it is still happening, even after I apply Eric's patch.

```
------------[ cut here ]------------
WARNING: CPU: 0 PID: 22534 at net/mptcp/protocol.c:764 mptcp_reset_timer+0x12a/0x160 net/mptcp/protocol.c:764
Modules linked in:
CPU: 0 PID: 22534 Comm: kworker/0:1 Not tainted 5.11.0-rc4e667aa91249d564c5b47a4302002dd8876ea201d #67
Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS rel-1.13.0-0-gf21b5a4aeb02-prebuilt.qemu.org 04/01/2014
Workqueue: events mptcp_worker
RIP: 0010:mptcp_reset_timer+0x12a/0x160 net/mptcp/protocol.c:764
Code: e8 1b 2b ac ff e8 46 2e 83 fe 48 b8 00 00 00 00 00 fc ff df 48 c7 04 03 00 00 00 00 48 83 c4 40 5b 5d 41 5c c3 e8 26 2e 83 fe <0f> 0b 41 bc c8 00 00 00 eb 98 e8 c7 fb aa fe e9 30 ff ff ff 48 c7
RSP: 0018:ffffc9000050fc60 EFLAGS: 00010293
RAX: 0000000000000000 RBX: 1ffff920000a1f8c RCX: 0000000000000000
RDX: ffff88800fe58000 RSI: ffffffff82b2bfba RDI: 0000000000000003
RBP: ffff888034916000 R08: 0000000000000000 R09: 0000000000000004
R10: ffffffff82b2bf57 R11: 0000000000000007 R12: 0000000000000000
R13: ffff8880349165b8 R14: ffff88800fc88000 R15: 0000000000000004
FS:  0000000000000000(0000) GS:ffff88811b400000(0000) knlGS:0000000000000000
CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
CR2: 0000000000000000 CR3: 00000000179e9002 CR4: 0000000000170ef0
Call Trace:
 mptcp_worker+0xe47/0x1310 net/mptcp/protocol.c:2307
 process_one_work+0x896/0x1170 kernel/workqueue.c:2275
 worker_thread+0x605/0x1350 kernel/workqueue.c:2421
 kthread+0x344/0x410 kernel/kthread.c:292
 ret_from_fork+0x22/0x30 arch/x86/entry/entry_64.S:296
```

e667aa91249d ("Eric's fix")  (HEAD) (4 days ago) <Christoph Paasch>
b4875a2a7e4a ("some more debugging in sk_stream_kill_queues")  (4 days ago) <Christoph Paasch>
70856533a4d9 ("DO-NOT-MERGE: mptcp: enabled by default")  (tag: export/20210128T064336, mptcp_net-next/export) (4 days ago) <Matthieu Baerts>
a475850233b6 ("DO-NOT-MERGE: mptcp: add GitHub Actions")  (4 days ago) <Matthieu Baerts>
489c73c74130 ("DO-NOT-MERGE: mptcp: use kmalloc on kasan build")  (4 days ago) <Paolo Abeni>
3519fa325557 ("mptcp: add netlink event support")  (4 days ago) <Florian Westphal>
6db4b0635a33 ("genetlink: add CAP_NET_ADMIN test for multicast bind")  (4 days ago) <Florian Westphal>
6c950820c9c5 ("mptcp: avoid lock_fast usage in accept path")  (4 days ago) <Florian Westphal>
5c072ef9aed0 ("mptcp: pass subflow socket to a few helpers")  (4 days ago) <Florian Westphal>
2709acb7cea6 ("mptcp: split __mptcp_close_ssk helper")  (4 days ago) <Florian Westphal>
ad9540369a8f ("mptcp: move pm netlink work into pm_netlink")  (4 days ago) <Florian Westphal>
b77a832e09ee ("mptcp: pm: add lockdep assertions")  (4 days ago) <Florian Westphal>
3d2aa9057623 ("selftests: mptcp: add command line arguments for mptcp_join.sh")  (4 days ago) <Geliang Tang>
3105f3e7435b ("selftests: mptcp: add testcases for ADD_ADDR with port")  (4 days ago) <Geliang Tang>
53118be7c75a ("mptcp: add the mibs for ADD_ADDR with port")  (4 days ago) <Geliang Tang>
fe8cb87a14c8 ("selftests: mptcp: add port argument for pm_nl_ctl")  (4 days ago) <Geliang Tang>
e5190ee5ecc5 ("mptcp: deal with MPTCP_PM_ADDR_ATTR_PORT in PM netlink")  (4 days ago) <Geliang Tang>
6322ac0958d8 ("mptcp: enable use_port when invoke addresses_equal")  (4 days ago) <Geliang Tang>
631325c84980 ("mptcp: add port number check for MP_JOIN")  (4 days ago) <Geliang Tang>
c06d0861a72d ("mptcp: add a new helper subflow_req_create_thmac")  (4 days ago) <Geliang Tang>
99be95715a1d ("mptcp: drop unused skb in subflow_token_join_request")  (4 days ago) <Geliang Tang>
12a982e474d2 ("mptcp: create the listening socket for new port")  (4 days ago) <Geliang Tang>
9147162e1048 ("selftests: mptcp: add testcases for newly added addresses")  (4 days ago) <Geliang Tang>
4010c2da6bf6 ("selftests: mptcp: use minus values for removing address numbers")  (4 days ago) <Geliang Tang>
ff8b5021e022 ("mptcp: send ack for every add_addr")  (4 days ago) <Geliang Tang>
cc6454e76894 ("mptcp: create subflow or signal addr for newly added address")  (4 days ago) <Geliang Tang>
b37ae55190ee ("mptcp: drop *_max fields in mptcp_pm_data")  (4 days ago) <Geliang Tang>
8aa9be3366cc ("mptcp: use WRITE_ONCE/READ_ONCE for the pernet *_max")  (4 days ago) <Geliang Tang>
cae4ced4d374 ("bpf:selftests: add bpf_mptcp_sock() verifier tests")  (4 days ago) <Nicolas Rybowski>
7bd6ee3bc08e ("bpf:selftests: add MPTCP test base")  (4 days ago) <Nicolas Rybowski>
d36880e9f98e ("bpf: add 'bpf_mptcp_sock' structure and helper")  (4 days ago) <Nicolas Rybowski>
710910e564d3 ("bpf: expose is_mptcp flag to bpf_tcp_sock")  (4 days ago) <Nicolas Rybowski>
75a7ca5b9d49 ("linux: handle MPTCP consistently with TCP")  (4 days ago) <Paolo Abeni>
d1f3bdd4eaae ("net: dsa: rtl8366rb: standardize init jam tables")  (mptcp_net-next/net-next) (5 days ago) <Lorenzo Carletti>

CONFIG-file: 
[CONFIG.txt](https://github.com/multipath-tcp/mptcp_net-next/files/5905281/CONFIG.txt)

**Comment by cpaasch** [comment]
Still happening on:

d82d76887ec6 ("mptcp: fix spurious retransmissions")  (HEAD) (8 hours ago) <Paolo Abeni>
9c23f272d8c2 ("Eric's fix")  (8 hours ago) <Christoph Paasch>
e2fe9491e272 ("DO-NOT-MERGE: mptcp: enabled by default")  (tag: export/20210202T061758, mptcp_net-next/export) (18 hours ago) <Matthieu Baerts>
13b4d63ddf90 ("DO-NOT-MERGE: mptcp: add GitHub Actions")  (18 hours ago) <Matthieu Baerts>
c2c68443db36 ("DO-NOT-MERGE: mptcp: use kmalloc on kasan build")  (18 hours ago) <Paolo Abeni>
6399d644bb8c ("mptcp: add netlink event support")  (18 hours ago) <Florian Westphal>
cf6cac4fe8e0 ("genetlink: add CAP_NET_ADMIN test for multicast bind")  (18 hours ago) <Florian Westphal>
875cda994e71 ("mptcp: avoid lock_fast usage in accept path")  (18 hours ago) <Florian Westphal>
d30f162ea307 ("mptcp: pass subflow socket to a few helpers")  (18 hours ago) <Florian Westphal>
5ee34d506df6 ("mptcp: split __mptcp_close_ssk helper")  (18 hours ago) <Florian Westphal>
fbc8817fc673 ("mptcp: move pm netlink work into pm_netlink")  (18 hours ago) <Florian Westphal>
77a274c37515 ("mptcp: pm: add lockdep assertions")  (18 hours ago) <Florian Westphal>
e75bbfc308c5 ("selftests: mptcp: add command line arguments for mptcp_join.sh")  (18 hours ago) <Geliang Tang>
7b74dee5eaea ("selftests: mptcp: add testcases for ADD_ADDR with port")  (18 hours ago) <Geliang Tang>
7dad582f33a9 ("mptcp: add the mibs for ADD_ADDR with port")  (18 hours ago) <Geliang Tang>
fc67ce163233 ("selftests: mptcp: add port argument for pm_nl_ctl")  (18 hours ago) <Geliang Tang>
4864e76b5c79 ("mptcp: deal with MPTCP_PM_ADDR_ATTR_PORT in PM netlink")  (18 hours ago) <Geliang Tang>
b99b4c0f0df6 ("mptcp: enable use_port when invoke addresses_equal")  (18 hours ago) <Geliang Tang>
6f9d0f9b169e ("mptcp: add port number check for MP_JOIN")  (18 hours ago) <Geliang Tang>
6f2398c5cfbb ("mptcp: add a new helper subflow_req_create_thmac")  (18 hours ago) <Geliang Tang>
28985de27e78 ("mptcp: drop unused skb in subflow_token_join_request")  (18 hours ago) <Geliang Tang>
47c71c622d8d ("mptcp: create the listening socket for new port")  (18 hours ago) <Geliang Tang>
b8a22e0eddec ("selftests: mptcp: add testcases for newly added addresses")  (18 hours ago) <Geliang Tang>
5978f842807e ("selftests: mptcp: use minus values for removing address numbers")  (18 hours ago) <Geliang Tang>
b8cda5afe5f6 ("mptcp: send ack for every add_addr")  (18 hours ago) <Geliang Tang>
8333d08c38ec ("mptcp: create subflow or signal addr for newly added address")  (18 hours ago) <Geliang Tang>
1031d4b86260 ("mptcp: drop *_max fields in mptcp_pm_data")  (18 hours ago) <Geliang Tang>
bb2d3335f171 ("mptcp: use WRITE_ONCE/READ_ONCE for the pernet *_max")  (18 hours ago) <Geliang Tang>
e2868c035bc2 ("bpf:selftests: add bpf_mptcp_sock() verifier tests")  (18 hours ago) <Nicolas Rybowski>
c6813456dd5b ("bpf:selftests: add MPTCP test base")  (18 hours ago) <Nicolas Rybowski>
0bb29bf4e878 ("bpf: add 'bpf_mptcp_sock' structure and helper")  (18 hours ago) <Nicolas Rybowski>
bc464f007527 ("bpf: expose is_mptcp flag to bpf_tcp_sock")  (18 hours ago) <Nicolas Rybowski>
d4a677fa8940 ("linux: handle MPTCP consistently with TCP")  (18 hours ago) <Paolo Abeni>
2c877743ac45 ("mptcp: fix length of MP_PRIO suboption")  (18 hours ago) <Davide Caratti>
9ae4bdc6e4c1 ("Merge branch 'rework-the-memory-barrier-for-scrq-entry'")  (mptcp_net-next/net-next) (20 hours ago) <Jakub Kicinski>

CONFIG:
[CONFIG.txt](https://github.com/multipath-tcp/mptcp_net-next/files/5915306/CONFIG.txt)

**Comment by matttbe** [comment]
Should be fixed thanks to a new patch from Paolo: f880caea34bf: mptcp: fix spurious retransmissions
